### ISSUE TITLE
Missing option chain data

### ISSUE THREAD

**Issue by bigiman** [ranaroussi/yfinance#1847]
### Describe bug

Missing actual option chain data.

### Simple code that reproduces your problem

``` python
import yfinance as yf

msft = yf.Ticker("MSFT")

opt = msft.options
print(opt)
opt = msft.option_chain('2024-12-20')
print(opt)
```

### Debug log

```
Traceback (most recent call last):
...
  File "/Users/michal/prj/OCM_prj/aaa.py", line 14, in <module>
    opt = msft.option_chain('2024-12-20')
  File "/usr/local/lib/python3.9/site-packages/yfinance/ticker.py", line 99, in option_chain
    raise ValueError(
ValueError: Expiration `2024-12-20` cannot be found. Available expiration are: []
```

### Bad data proof

![Bildschirmfoto 2024-02-01 um 22 15 16](https://github.com/ranaroussi/yfinance/assets/86003083/46699505-f38f-4970-88e2-c1f1049b94bc)

### `yfinance` version

Version: 0.2.36

### Python version

3.11

### Operating system

Mac OS

**Comment by ValueRaider** [comment]
That's not debug log. Instructions in new issue form.

**Comment by bigiman** [comment]
> That's not debug log. Instructions in new issue form.

Sorry, don't understand what exactly do you mean. If that's not debug log in the form you expect, then I apologise. The code I provided is the basic code for retrieving options chain data which is simply not working.

**Comment by 46martin11** [comment]
I do agree with bigiman. Since about 2 weeks ago the option expiration list is empty (so when you call .option on a ticker the list of expirations that gets returned is an empty list). This was working prior to the 21 January update

**Comment by ValueRaider** [comment]
![yf-issue-dbg-log](https://github.com/ranaroussi/yfinance/assets/96923577/5931ab98-b78b-417b-8e31-80e0e65090b4)

I don't know how to make more explicit. This issue is not common so not easily reproducible, we need the debug log.

**Comment by bigiman** [comment]
This issue is incredibly easy to reproduce. But OK, I added the error log which I get in debug mode. I hope this is what you were missing.

**Comment by ValueRaider** [comment]
> This issue is incredibly easy to reproduce.

For you but not for me:
```
>>> msft = yf.Ticker("MSFT")
>>> 
>>> opt = msft.options

>>> print(opt)
('2024-02-16', '2024-02-23', '2024-03-01', '2024-03-08', '2024-03-15', '2024-03-22', '2024-04-19', '2024-05-17', '2024-06-21', '2024-07-19', '2024-08-16', '2024-09-20', '2024-12-20', '2025-01-17', '2025-06-20', '2025-12-19', '2026-01-16', '2026-06-18', '2026-12-18')
>>> opt = msft.option_chain('2024-12-20')
>>> print(opt)
Options(calls=         contractSymbol             lastTradeDate  strike  lastPrice     bid     ask  change  percentChange  volume  openInterest  impliedVolatility  inTheMoney contractSize currency
0   MSFT241220C00125000 2024-01-26 16:01:42+00:00   125.0     284.00  296.15  300.50    0.00       0.000000     1.0            48           0.763552        True      REGULAR      USD
...
```

I'm still not seeing the debug log. You need to run `yf.enable_debug_mode()` first.

**Comment by ValueRaider** [comment]
Also, this doesn't line up with latest source code:
```
File "/usr/local/lib/python3.9/site-packages/yfinance/ticker.py", line 99, in option_chain
    raise ValueError(
```

That should be line 90 not 99. I don't think you have latest version.

**Comment by bigiman** [comment]
I'm lost.
AttributeError: module 'yfinance' has no attribute 'enable_debug_mode'

pip show yfinance gives:
Name: yfinance
Version: 0.2.36
Summary: Download market data from Yahoo! Finance API
Home-page: https://github.com/ranaroussi/yfinance
Author: Ran Aroussi
Author-email: ran@aroussi.com
License: Apache
Location: /usr/local/lib/python3.11/site-packages
Requires: appdirs, beautifulsoup4, frozendict, html5lib, lxml, multitasking, numpy, pandas, peewee, pytz, requests
Required-by:

**Comment by ValueRaider** [comment]
PIP is wrong. Read the 'new issue' instructions again.

**Comment by bigiman** [comment]
import yfinance as yf
print(yf.__version__)

0.1.62

You are right. Some other dependencies denied to install the latest version....
If I tried it in virtual environment, it worked.

Thanks, have to investigate....

**Comment by ValueRaider** [comment]
How can the new issue form be improved to avoid a repeat of this conversation?

**Comment by bigiman** [comment]
Older version of pandas made troubles in my case.

I mean it would be better if pip worked properly

Now you have already improved it with this 😉
!!! IMPORTANT !!! FOLLOW THESE INSTRUCTIONS CAREFULLY !!!

**Comment by bigiman** [comment]
Solved with upgrade to higher version of pandas library. 
yinance 0.2.36 requires pandas at least 1.3.0

**Comment by ValueRaider** [comment]
What about getting people to provide debug log?

**Comment by bigiman** [comment]
If they use older version, debug log is not present (that was my case).
I would move the info about debug log closer to the beginning of the instructions. Some small example would help as well.

**Comment by raphi6** [comment]
I have the same issue right now, however there appears to be no S&P500 options data on the Yahoo! webpage either: https://finance.yahoo.com/quote/%5EGSPC/options/

Is this a temporary problem?
